java/integTest/src/test/java/glide/standalone/TransactionTests.java

```diff
@@ -307,6 +308,62 @@ public void watch() {
         assertInstanceOf(RequestException.class, executionException.getCause());
     }
 
+    @Test
+    @SneakyThrows
+    public void watch_binary() {
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String key3 = "{key}-3" + UUID.randomUUID();
+        String key4 = "{key}-4" + UUID.randomUUID();
+        String foobarString = "foobar";
+        String helloString = "hello";
+        GlideString[] keys =
+                new GlideString[] {GlideString.gs(key1), GlideString.gs(key2), GlideString.gs(key3)};
+        Transaction setFoobarTransaction = new Transaction();
+        Transaction setHelloTransaction = new Transaction();
+        String[] expectedExecResponse = new String[] {OK, OK, OK};
+
+        // Returns null when a watched key is modified before it is executed in a transaction command.
+        // Transaction commands are not performed.
+        assertEquals(OK, client.watch(keys).get());
+        assertEquals(OK, client.set(key2, helloString).get());
+        setFoobarTransaction.set(key1, foobarString).set(key2, foobarString).set(key3, foobarString);
+        assertEquals(null, client.exec(setFoobarTransaction).get());
+        assertEquals(null, client.get(key1).get()); // Sanity check
+        assertEquals(helloString, client.get(key2).get());
+        assertEquals(null, client.get(key3).get());
```

Comment:
nit: (I know that initial test uses equals)
```suggestion
        assertNull(client.exec(setFoobarTransaction).get());
        assertNull(client.get(key1).get()); // Sanity check
        assertEquals(helloString, client.get(key2).get());
        assertNull(client.get(key3).get());
```